### PR TITLE
Markdown partials

### PR DESCRIPTION
**Pull request by yorhodes**

### CODE REVIEW

File: sync-partials.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import glob from "glob";
```
Comment (nambrot) [inline sync-partials.js:1]: Since when we are javascript-over-typescript people?

File: sync-partials.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import fs from "fs";
+
+// use this comment syntax to include markdown partials between the include and end tags
+// <!-- INCLUDE path-to-partial.md -->
+// <!-- END -->
+
+const includeRegex = /(\<\!\-\- INCLUDE (.*.md) -->)(.|\n)*(<!-- END -->)/g;
+const warning =
```
Comment (nambrot) [inline sync-partials.js:9]: Does this warning show up on gitbook?

File: sync-partials.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  "<!-- WARNING: copied from the included markdown path. Do not edit directly. -->";
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_the_replacement
+function replacer(_, includeTag, markdownFile, __, endTag) {
+  const content = fs.readFileSync(markdownFile, "utf8");
+  return `${includeTag}\n${warning}\n${content}\n${warning}\n${endTag}`;
```
Comment (nambrot) [inline sync-partials.js]: This is like the nittiest thing ever, but wdyt of 
```suggestion
  return [includeTag, content, warning, endTag].join('\n');
```

File: package.json [hunk 4]
```diff
@@ -15,7 +16,7 @@
     "private": true,
     "type": "module",
     "scripts": {
-        "sync": "node sync-addresses.js > ./resources/addresses.md",
+        "sync": "node sync-addresses.js > ./resources/addresses.md && node sync-partials.js",
```
Comment (nambrot) [inline package.json:22]: Should we run this in a commit hook?

File: sync-partials.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import glob from "glob";
+import fs from "fs";
+
+// use this comment syntax to include markdown partials between the include and end tags
+// <!-- INCLUDE path-to-partial.md -->
+// <!-- END -->
+
+const includeRegex = /(\<\!\-\- INCLUDE (.*.md) -->)(.|\n)*(<!-- END -->)/g;
```
Comment (nambrot) [inline sync-partials.js]: Why have the end tag?

File: sync-partials.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+const warning =
+  "<!-- WARNING: copied from the included markdown path. Do not edit directly. -->";
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_the_replacement
+function replacer(_, includeTag, markdownFile, __, endTag) {
+  const content = fs.readFileSync(markdownFile, "utf8");
```
Comment (nambrot) [inline sync-partials.js]: What about recursive/nested partials?

Comment (nambrot) [reply sync-partials.js]: jk jk

File: package.json (hunk 4 above)
Comment (nambrot) [reply package.json:22]: Or should we run this in CI and assert there is no diff?

Or do that in a cron job?

File: README.md [hunk 7]
```diff
@@ -18,46 +18,8 @@ Hyperlane's [APIs](./#hyperlane-apis) offer developers secure and simple ways to
...
-    style aws fill:#fc9,stroke:#f90,stroke-width:3px
-    style rel fill:#e4f3ff,stroke:#025aa1,stroke-width:3px 
-    style oc fill:#c8e6fe,stroke:#ddd,stroke-dasharray: 5 5,stroke-width:3px
-    style rc fill:#c8e6fe,stroke:#ddd,stroke-dasharray: 5 5,stroke-width:3px
-    style sc stroke-width:3px
-    style rec stroke-width:3px
-    style mb fill:#e4f3ff,stroke:#025aa1,stroke-width:3px
-    style mb2 fill:#e4f3ff,stroke:#025aa1,stroke-width:3px
-```
+<!-- INCLUDE diagrams/messaging-simple.md -->
```
Comment (nambrot) [inline README.md:21]: I assume there is a separate PR for this, since presumably this would break right now?

**Review by nambrot — APPROVED** [review]
TBH, not as bad as I thought it would be. Actually kinda cool. Still a rabbit hole tho :P

File: sync-partials.js (hunk 2 above)
Comment (yorhodes) [reply sync-partials.js:9]: no

File: sync-partials.js (hunk 5 above)
Comment (yorhodes) [reply sync-partials.js]: we need to know where the content ends so we can overwrite *idempotently*

File: README.md (hunk 7 above)
Comment (yorhodes) [reply README.md:21]: base of this pr is `diagrams` branch with the content

**Pull request merged**